recipes/libuuid/all/patches/0001-check-sys-syscall-h+wrap-includes.patch

```diff
@@ -0,0 +1,36 @@
+--- configure.ac
++++ configure.ac
+@@ -20,7 +20,7 @@ AC_SEARCH_LIBS([socket], [socket],
+ AC_SUBST([SOCKET_LIBS])
+ 
+ # Checks for header files.
+-AC_CHECK_HEADERS([fcntl.h inttypes.h limits.h netinet/in.h stdlib.h string.h sys/file.h sys/ioctl.h sys/socket.h sys/time.h unistd.h])
++AC_CHECK_HEADERS([fcntl.h inttypes.h limits.h netinet/in.h stdlib.h string.h sys/file.h sys/ioctl.h sys/socket.h sys/syscall.h sys/time.h unistd.h])
+ AC_CHECK_DECLS([_SC_HOST_NAME_MAX])
+ 
+ # Checks for typedefs, structures, and compiler characteristics.
+--- randutils.c
++++ randutils.c
+@@ -6,14 +6,22 @@
+  * This file may be redistributed under the terms of the
+  * GNU Lesser General Public License.
+  */
++
++#include "config.h"
```

Comment:
It is typical in an Autoconf build to define HAVE_CONFIG_H on the command line, and then protect any include of "config.h" with that macro.
Had I noticed that was not there already, I would have added that too - thanks!

Reply:
Haha, let's do that!
Btw, I checked the other sources:
`gen_uuid.c` also uses: `HAVE_UNISTD_H`, `HAVE_STDLIB_H`, ... and does not include `"config.h"`.
Looks like a bug to me.

I'm going to replace the `#include "config.h"` with `#include "c.h"`.
The tarball also contains `uuidP.h` and `c.h`.
The first is a private uuid header, the second included `config.h` + more.

Reply:
The one thing we have to be careful of is that we don't use config.h or the macros in any public header files.  Other than that, adding the macros should only help.  I'm also of the opinion that it's a BadThing(TM) to depend on a header including another header  because that  is an implementation detail.

Reply:
Indeed.
I just took a glance at a log and only `uuid.h` is installed.
So I guess it's fine to modify/use the private `c.h`/`uuidP.h` headers.

Reply:
Thanks for the dudilgence!